src/python.rs

```diff
@@ -556,17 +556,42 @@ impl<'p> Python<'p> {
         FromPyPointer::from_borrowed_ptr_or_opt(self, ptr)
     }
 
-    /// Lets the Python interpreter check for pending signals and invoke the
-    /// corresponding signal handlers. This can run arbitrary Python code.
+    /// Lets the Python interpreter check and handle any pending signals and invoke the
+    /// corresponding signal handlers.
     ///
-    /// If an exception is raised by the signal handler, or the default signal
-    /// handler raises an exception (such as `KeyboardInterrupt` for `SIGINT`),
-    /// an `Err` is returned.
+    /// Returns `Err(PyErr)` if any signal handler raises an exception.
     ///
-    /// This is a wrapper of the C function `PyErr_CheckSignals()`. It is good
-    /// practice to call this regularly in a long-running calculation since
-    /// SIGINT and other signals handled by Python code are left pending for its
-    /// entire duration.
+    /// These signals include `SIGINT` (normally raised by CTRL + C), which by default raises
+    /// `KeyboardInterrupt`. For this reason it is good practice to call this function regularly
+    /// if you have a long-running Rust function so that your users can cancel it.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use pyo3::prelude::*;
+    /// # fn main(){
```

Comment:
Oh, nice.

Reply:
Actually, I'd prefer to leave them in, because the invisible `main` function is copied when using the book's clipboard button, as well as by the "Run" button (for crates with a playground link). This saves users from having to manually put in a main function.

Reply:
Is pyo3 even usable in the playground?!

Reply:
:+1:

Reply:
> But the playground link can be customized, so in principle we could host our own playground. (We'd need to host a python interpreter too? How does that work?)

Interesting question - at least a libpython would need to be available. AFAIU the playground deployment is container based, so the used container can hopefully be easily amended.

But I don't know who would host it...